lms/djangoapps/course_blocks/transformers/user_partitions.py

```diff
@@ -79,16 +84,34 @@ def transform_block_filters(self, usage_info, block_structure):
             return [block_structure.create_universal_filter()]
 
         user_groups = get_user_partition_groups(usage_info.course_key, user_partitions, user, 'id')
-        group_access_filter = block_structure.create_removal_filter(
-            lambda block_key: not (
-                has_access(user, 'staff', block_key) or
-                block_structure.get_transformer_block_field(block_key, self, 'merged_group_access').check_group_access(
-                    user_groups
+        if self.hide_access_denials:
+            group_access_filter = block_structure.create_removal_filter(
+                lambda block_key: not (
+                    has_access(user, 'staff', block_key) or
+                    block_structure.get_transformer_block_field(
+                        block_key, self, 'merged_group_access'
+                    ).check_group_access(
+                        user_groups
+                    )
                 )
             )
-        )
-
-        result_list.append(group_access_filter)
+            result_list.append(group_access_filter)
+        else:
+            for block_key in block_structure.topological_traversal():
+                if not (
+                        has_access(user, 'staff', block_key) or
+                        block_structure.get_transformer_block_field(
+                            block_key, self, 'merged_group_access'
+                        ).check_group_access(
+                            user_groups
+                        )):
+                    block_structure.override_xblock_field(
+                        block_key, 'authorization_denial_reason', 'feature_based_enrollments'
```

Comment:
This should be more general. In particular, it should be based on the UserPartition that is limiting the content. The message should be computed by calling `UserPartition.access_denied_message`

Reply:
It should probably also just filter out the content if there's no `access_denied_message` available.

Reply:
why would an access_denied_message not be available?  
if it is because of a situation where a partition doesn't have an access_denied_message defined, but has a name, could we just filter them out in the access_denied_filter transformer?  
that might make more sense because we're already checking both values and filtering out blocks there

Reply:
Not all partitions generate a message when they deny access (in fact, basically none of them do, right now)